Code/GraphMol/Chirality.cpp

```diff
@@ -451,65 +451,141 @@ bool atomIsCandidateForRingStereochem(const ROMol &mol, const Atom *atom) {
+
+  for (ROMol::AtomIterator ait = mol.beginAtoms(); ait != mol.endAtoms();
+       ++ait) {
+    const Atom *atom = *ait;
+    if (atomsSeen[atom->getIdx()]) continue;
+    if (atom->getChiralTag() == Atom::CHI_UNSPECIFIED ||
+        atom->hasProp(common_properties::_CIPCode) ||
+        !mol.getRingInfo()->numAtomRings(atom->getIdx()) ||
+        !atomIsCandidateForRingStereochem(mol, atom)) {
+      continue;
+    }
+    // do a BFS from this ring atom along ring bonds and find other
+    // stereochemistry candidates.
+    std::list<const Atom *> nextAtoms;
+    // start with finding viable neighbors
+    ROMol::OEDGE_ITER beg, end;
+    boost::tie(beg, end) = mol.getAtomBonds(atom);
+    while (beg != end) {
+      unsigned int bidx = mol[*beg]->getIdx();
+      if (!bondsSeen[bidx]) {
+        bondsSeen.set(bidx);
+        if (mol.getRingInfo()->numBondRings(bidx)) {
+          const Atom *oatom = mol[*beg]->getOtherAtom(atom);
+          if (!atomsSeen[oatom->getIdx()]) {
+            nextAtoms.push_back(oatom);
+            atomsUsed.set(oatom->getIdx());
+          }
+        }
+      }
+      ++beg;
+    }
     INT_VECT ringStereoAtoms(0);
-    atom->getPropIfPresent(common_properties::_ringStereoAtoms,
-                           ringStereoAtoms);
-
-    const VECT_INT_VECT atomRings = ringInfo->atomRings();
-    for (VECT_INT_VECT::const_iterator ringIt = atomRings.begin();
-         ringIt != atomRings.end(); ++ringIt) {
-      if (std::find(ringIt->begin(), ringIt->end(),
-                    static_cast<int>(atom->getIdx())) != ringIt->end()) {
-        for (INT_VECT::const_iterator idxIt = ringIt->begin();
-             idxIt != ringIt->end(); ++idxIt) {
-          int same = 1;
-          if (*idxIt != static_cast<int>(atom->getIdx()) &&
-              mol.getAtomWithIdx(*idxIt)->getChiralTag() !=
-                  Atom::CHI_UNSPECIFIED &&
-              !mol.getAtomWithIdx(*idxIt)->hasProp(
-                  common_properties::_CIPCode) &&
-              atomIsCandidateForRingStereochem(mol,
-                                               mol.getAtomWithIdx(*idxIt))) {
-            // we get to keep the stereochem specification on this atom:
-            if (mol.getAtomWithIdx(*idxIt)->getChiralTag() !=
-                atom->getChiralTag()) {
-              same = -1;
+    if (!nextAtoms.empty()) {
+      atom->getPropIfPresent(common_properties::_ringStereoAtoms,
+                             ringStereoAtoms);
+    }
+
+    while (!nextAtoms.empty()) {
+      const Atom *ratom = nextAtoms.front();
+      nextAtoms.pop_front();
+      atomsSeen.set(ratom->getIdx());
+      if (ratom->getChiralTag() != Atom::CHI_UNSPECIFIED &&
+          !ratom->hasProp(common_properties::_CIPCode) &&
+          atomIsCandidateForRingStereochem(mol, ratom)) {
+        int same = (ratom->getChiralTag() == atom->getChiralTag()) ? 1 : -1;
+        ringStereoAtoms.push_back(same * (ratom->getIdx() + 1));
```

Comment:
As far as I can tell, this tagging is never actually used, i.e. you always push the tagged index but never actually use the fact that the chiral tags are the same or different.  All I can find is unpacking the tag to the proper atom index.